src/app/api/advocates/route.ts

```diff
@@ -1,12 +1,36 @@
-import db from "../../../db";
-import { advocates } from "../../../db/schema";
-import { advocateData } from "../../../db/seed/advocates";
+  import db from "../../../db";
+import {advocates} from "../../../db/schema";
+import {ilike, or, sql} from "drizzle-orm";
 
-export async function GET() {
-  // Uncomment this line to use a database
-  // const data = await db.select().from(advocates);
+const PAGE_LIMIT = 5;
 
-  const data = advocateData;
+export async function GET(req: Request) {
+  const url = new URL(req.url);
+  const query = url.searchParams.get("query") || "";
+  const page = parseInt(url.searchParams.get("page") || "1", 10);
+  const offset = (page - 1) * PAGE_LIMIT;
+  const where = ilike(advocates.contentSearch, `%${query}%`);
 
-  return Response.json({ data });
-}
+  const items = await db
+    .select()
+    .from(advocates)
+    .where(where)
+    .limit(PAGE_LIMIT)
+    .offset(offset);
+
+  // Fetch total count for metadata
+  const totalCountResult = await db
+    .select({count: sql`COUNT(id)`})
+    .from(advocates)
+    .where(where);
+
+  const totalCount = totalCountResult[0]?.count || 0;
+
+  return Response.json({
+    items,
+    meta: {
+      currentPage: page,
+      totalPages: Math.ceil(totalCount / PAGE_LIMIT),
+    },
+  });
```

Comment:
we could define return type from "types" folder. And also wrap in some generic "pagination" function, it is likely not the first and last time we are using pagination